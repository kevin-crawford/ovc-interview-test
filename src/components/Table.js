import React, { Component } from "react";
import { connect } from "react-redux";
import Cell from "./Cell";
import getData from "../redux/actions/dataActions";

export class Table extends Component {
  componentDidMount() {
    this.props.getData();
  }

  render() {
    const { headings, user } = this.props;

    const theadMarkup = (
      <tr key="heading">
        {headings.map((heading, index) => (
          <Cell key={`heading-${index}`} content={heading} header={true} />
        ))}
      </tr>
    );

    const tbodyMarkup = user.data.map((user, index) => (
      <tr key={`row-${index}`}>
        <td>{user.name}</td>
        <td>
          <a href={`mailto:${user.email}`}>{user.email}</a>
        </td>
        <td>{user.address.city}</td>
        <td>{user.company.name}</td>
      </tr>
    ));

    return (
      <table className="Table">
        <thead>{theadMarkup}</thead>
        <tbody>{tbodyMarkup}</tbody>
      </table>
    );
  }
}

// map our redux store/state to the component props
const mapStateToProps = state => ({
  user: state.data
});

export default connect(
  mapStateToProps,
  { getData }
)(Table);
