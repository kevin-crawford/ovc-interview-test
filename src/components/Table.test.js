import React from "react";
import { Provider } from "react-redux";
import { mount, shallow } from "enzyme";
import { Table } from "./Table";

import thunk from "redux-thunk";
import configureStore from "redux-mock-store";

const mockStore = configureStore([thunk]);

describe("<Table />", () => {
  it("Renders without crashing", () => {
    const store = mockStore({});
    const wrapper = shallow(
      <Provider store={store}>
        <Table />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
