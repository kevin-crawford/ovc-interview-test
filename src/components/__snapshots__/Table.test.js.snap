// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> Renders without crashing 1`] = `ShallowWrapper {}`;
